packages/client/hooks/usePortal.tsx

```diff
@@ -33,7 +33,7 @@ const getParent = (parentId: string | undefined) => {
 const usePortal = (options: UsePortalOptions = {}) => {
   const portalRef = useRef<HTMLDivElement>()
   const originRef = useRef<HTMLElement>()
-  const timeoutRef = useRef<number | null>(null)
+  const timeoutRef = useRef<number | NodeJS.Timeout | null>(null)
```

Comment:
I'm not sure what's the right thing to do here because I'm pretty sure this code only runs in the browser (or am I wrong?). If that's the case then the TS is wrong here and the only way to let TS know we want the browser version of set timeout is to use `window.setTimeout` directly which is pretty dumb. Anyway, I'm sharing my thoughts here 😅

Reply:
`timeoutRef` was causing a typecheck issue but it looks like upgrading to v4.4.2 has fixed this so I've reverted `usePortal` back to the way it was